### PR TITLE
Client: History and Details menus

### PR DESCRIPTION
**Pull request by psiinon**
## Overview
Add context menus for copying table cell values from the Client Details and History tabs.

## Related Issues
Fixes https://github.com/zaproxy/zaproxy/issues/8173

## Checklist
- [x] Update help
- [x] Update changelog
- [x] Run `./gradlew spotlessApply` for code formatting
- [ ] Write tests
- [ ] Check code coverage
- [x] Sign-off commits
- [x] Squash commits
- [x] Use a descriptive title

For more details, please refer to the [developer rules and guidelines](https://www.zaproxy.org/docs/developer/dev-rules-and-guidelines/).

### CODE REVIEW

File: addOns/client/src/main/resources/org/zaproxy/addon/client/resources/Messages.properties [hunk 1]
```diff
@@ -8,8 +8,16 @@ client.components.type.input = Input
...
+client.details.popup.copy.hrefs = Copy HREFs to Clipboard
+client.details.popup.copy.ids = Copy IDs to Clipboard
+client.details.popup.copy.texts = Copy Texts to Clipboard
 client.details.title = Client Details
 client.error.nofile = File not found : {0}
+client.history.popup.copy.nodeids = Copy Node IDs to Clipboard
+client.history.popup.copy.nodenames = Copy Node Names to Clipboard
+client.history.popup.copy.texts = Copy Texts to Clipboard
+client.history.popup.copy.types = Copy Types to Clipboard
+client.history.popup.copy.urls = Copy Source URLs to Clipboard
```
Comment (kingthorin) [inline addOns/client/src/main/resources/org/zaproxy/addon/client/resources/Messages.properties:20]: For the help too, do we really need to say "to Clipboard"?

File: addOns/client/src/main/java/org/zaproxy/addon/client/PopupMenuClientDetailsCopy.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+    @Override
+    public void performAction(ActionEvent e) {
+        StringBuilder sb = new StringBuilder();
+        for (ClientSideComponent obj : getClientDetailsPanel().getSelectedRows()) {
+            String val = this.function.apply(obj);
+            if (val != null) {
+                sb.append(val);
+            }
+            sb.append('\n');
```
[the remark this answers is not in the document]
Comment (kingthorin) [reply addOns/client/src/main/java/org/zaproxy/addon/client/PopupMenuClientDetailsCopy.java:48]: `System.lineSeparator()`

File: addOns/client/src/main/java/org/zaproxy/addon/client/PopupMenuItemClientHistory.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        this.addActionListener(l -> performAction(l));
+    }
+
+    @Override
+    public boolean isEnableForComponent(Component invoker) {
+        boolean enabled =
+                invoker instanceof ZapTable
+                        && ClientHistoryPanel.CLIENT_HISTORY_NAME.equals(invoker.getName());
+        this.setEnabled(!clientHistoryPanel.getSelectedRows().isEmpty());
+        return enabled;
```
[the remark this answers is not in the document]
Comment (psiinon) [reply addOns/client/src/main/java/org/zaproxy/addon/client/PopupMenuItemClientHistory.java:44]: ditto :)

File: addOns/client/src/main/resources/org/zaproxy/addon/client/resources/Messages.properties (hunk 1 above)
Comment (psiinon) [reply addOns/client/src/main/resources/org/zaproxy/addon/client/resources/Messages.properties:20]: Thats what we do in the core, so for consistancy "yes" 😁

**Comment by psiinon** [comment]
Updated

File: addOns/client/src/main/java/org/zaproxy/addon/client/PopupMenuClientDetailsCopy.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (thc202) [reply addOns/client/src/main/java/org/zaproxy/addon/client/PopupMenuClientDetailsCopy.java]: We are not using the system dependent separator in other places (e.g. `PopupMenuCopyUrls`).

[the remark this answers is not in the document]
Comment (kingthorin) [reply addOns/client/src/main/java/org/zaproxy/addon/client/PopupMenuClientDetailsCopy.java]: Shall we? I'm happy to tweak/adapt

[the remark this answers is not in the document]
Comment (thc202) [reply addOns/client/src/main/java/org/zaproxy/addon/client/PopupMenuClientDetailsCopy.java:48]: For the record I prefer always use the same.

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Review by kingthorin — APPROVED** [review]

**Comment by kingthorin** [comment]
FYI I created an issue for myself to remove "to Clipboard" from extensions/core/help. I'll try to tackle that sometime this week.

**Comment by thc202** [comment]
Sounds good but would be better to create one in the main repo (for the record and decision).

**Comment by kingthorin** [comment]
Sure, will do.

**Pull request merged**